projects/plugins/jetpack/class.jetpack.php

```diff
@@ -3873,52 +3881,61 @@ public function deactivate_dialog( $hook ) {
 			&& self::is_connection_ready()
 		) {
 
-			$active_plugins_using_connection = Connection_Plugin_Storage::get_all();
-
-			if ( count( $active_plugins_using_connection ) > 1 ) {
+			// Register jp-tracks-functions dependency.
+			Tracking::register_tracks_functions_scripts( true );
 
-				add_thickbox();
+			// add a deactivation script that will pick up deactivation actions for the Jetpack plugin.
+			$script_deps_path    = JETPACK__PLUGIN_DIR . '_inc/build/plugins-page.asset.php';
+			$script_dependencies = array( 'wp-polyfill' );
+			if ( file_exists( $script_deps_path ) ) {
+				$asset_manifest      = include $script_deps_path;
+				$script_dependencies = $asset_manifest['dependencies'];
+			}
 
-				// Register jp-tracks-functions dependency.
-				Tracking::register_tracks_functions_scripts();
+			wp_enqueue_script(
```

Comment:
This may be a good spot to use the new `register_script` method we've added to the Assets package:
https://github.com/Automattic/jetpack/blob/99da2d724343e0c0de56f142dc4641c8620518bd/projects/packages/assets/src/class-assets.php#L273-L303

Reply:
Very cool, nice that this sorts out all the script dependencies auto-magically! :) Will update.